[ADD] Inventory: add allocation report 19.4 (#19153)
odoo/documentation: content/applications/inventory_and_mrp/inventory/warehouses_storage/reporting/allocation_report.rst

=================
Allocation report
=================

.. |SO| replace:: :abbr:`SO (sales order)`
.. |SOs| replace:: :abbr:`SOs (sales orders)`
.. |RfQ| replace:: :abbr:`RfQ (request for quotation)`
.. |PO| replace:: :abbr:`PO (purchase order)`
.. |POs| replace:: :abbr:`POs (purchase orders)`
.. |MO| replace:: :abbr:`MO (manufacturing order)`
.. |MOs| replace:: :abbr:`MOs (manufacturing orders)`

In Odoo, users can use allocation reports to manually reserve units from incoming stock (either

> purchased or manufactured) to outgoing demands, such as sales orders (SOs) or components on

Reviewer: ```suggestion
purchased or manufactured) for outgoing demands, such as sales orders (SOs) or components on
```

demands.

The following documentation covers the process of using an allocation report to manually assign
incoming units of a product from a purchase order (PO) or a manufacturing order (MO) to fulfill
either an |SO| containing the purchased product or an |MO| with the product as a component.

.. _inventory/allocation_report/configuration:

Configuration
=============

By default, allocation reports are automatically enabled for incoming moves (e.g., receipts
generated from |POs|). However, for some operations, users must manually enable the feature.

To do so, navigate to :menuselection:`Inventory app --> Configuration --> Operations Types`. On the
dashboard, click the desired :guilabel:`Operation Type` (e.g., `Receipts` or `Manufacturing`), then
verify that the :guilabel:`Show Allocation` checkbox is enabled. Enabling this option automatically
displays the option to view an :ref:`allocation report
<inventory/allocation_report/workflow/report>` for the specific operation.

.. _inventory/allocation_report/workflow:

Workflow
========

The following outline summarizes all necessary steps for allocating a product using reception

> reports:

Reviewer: Just verifying -- did you mean reception reports or allocation reports here?

by purchasing it or by manufacturing it.

On the receipt from the |PO| or on the confirmed |MO|, users can open the allocation report, which
displays the incoming product. Next to the product is a list of all applicable outgoing transfers
(i.e., originating |SOs|/|MOs| containing the product) to which incoming units can be assigned.
Assigning these units does not move any inventory. Instead, it allows the user to flexibly choose
which demands to prioritize before actually receiving incoming units. When those units are received,
the user can fulfill the original demands.

.. note::
   If multiple products are replenished through the |PO|/|MO|, the allocation report lists all
   products with originating |SOs|/|MOs| available for reservation.

.. _inventory/allocation_report/workflow/confirm_demand:

Create and confirm demand
-------------------------

Using allocation reports for specific products requires either a previously confirmed |SO| with the
product or an |MO| where the product is used as a component. If no such record exists, create it.

.. important::
   To ensure Odoo can properly create allocations for purchased products, confirm that the selected

>    products have the :guilabel:`Purchase` option selected and its :guilabel:`Product Type` set to

Reviewer: ```suggestion
   products have the :guilabel:`Purchase` option selected and the :guilabel:`Product Type` set to
```
There was a subject/pronoun disagreement issue that I noted. I feel "the" would work best to resolve it without messing up the lines too much.

~~~~~~~~~~~~~~~~~~~~~~

To create a |PO|, navigate to the **Purchase** app and click :guilabel:`New` to create a new Request
for Quotation (RfQ). On the form, add the products in demand on the customers' |SOs| by clicking
:guilabel:`Add a product` in the *Products* tab and specifying the required :guilabel:`Quantity`.

After adding the products, click :guilabel:`Confirm Order` to turn the |RfQ| into a |PO|.

Replenish through |MO|
~~~~~~~~~~~~~~~~~~~~~~

To create an |MO|, navigate to :menuselection:`Manufacturing app --> Operations --> Manufacturing
Orders`. Then, click :guilabel:`New` to open a new |MO| form. Select the :guilabel:`Product` and
enter the :guilabel:`Quantity` to replenish.

Finally, :guilabel:`Confirm` the |MO|.

.. _inventory/allocation_report/workflow/report:

Allocating using the report
---------------------------

After confirming the replenishment for the product, open the allocation report to assign the pending
incoming stock to any available originating |SOs|/|MOs| requiring the product.

For confirmed |POs|, click the :icon:`fa-truck` :guilabel:`Receipt` smart button at the top of the
|PO| to open the receipt for the incoming stock of the product.

> Then, click the :icon:`fa-list` :guilabel:`Allocation` smart button appears at the top.

Reviewer: ```suggestion
Then, click the :icon:`fa-list` :guilabel:`Allocation` smart button at the top.
```

.. important::
   If the :guilabel:`Allocation` smart button does not appear for an |MO|, follow the
   :ref:`configuration process <inventory/allocation_report/configuration>` to enable it.

.. _inventory/allocation_report/workflow/report/dashboard:

Allocation report dashboard
~~~~~~~~~~~~~~~~~~~~~~~~~~~

The allocation report displays a list of available products on the replenishing |PO|/|MO|. Next to
each of these products is a list of originating |SOs|/|MOs| to which the products can be allocated.

Each row of this list displays the :guilabel:`Quantity` of reservable units out of the total demand
for the product. The quantity is highlighted if it cannot be reserved for the complete demand.

The toggle button in the :guilabel:`Reserved` column allows the user to :ref:`allocate
<inventory/allocation_report/workflow/report/allocate>` the specified quantity.

A link to each |SO|/|MO| is included in the :guilabel:`For` column, along with its expected delivery
:guilabel:`Date`.

.. note::
   |SOs|/|MOs| in the allocation report are listed in chronological order of their associated
   delivery or scheduled date (i.e., the |SO| with the oldest delivery order appears first or the

>    |MO| with the oldest scheduled date).

Reviewer: ```suggestion
   delivery or scheduled date (i.e., the |SO| with the oldest delivery order or the |MO| with the
   oldest scheduled date appears first).
```

=================
Allocation report
=================

.. |SO| replace:: :abbr:`SO (sales order)`
.. |SOs| replace:: :abbr:`SOs (sales orders)`
.. |RfQ| replace:: :abbr:`RfQ (request for quotation)`
.. |PO| replace:: :abbr:`PO (purchase order)`
.. |POs| replace:: :abbr:`POs (purchase orders)`
.. |MO| replace:: :abbr:`MO (manufacturing order)`
.. |MOs| replace:: :abbr:`MOs (manufacturing orders)`

In Odoo, users can use allocation reports to manually reserve units from incoming stock (either
purchased or manufactured) to outgoing demands, such as sales orders (SOs) or components on
manufacturing orders (MOs).

Odoo normally reserves existing stock to outgoing demands automatically based on their scheduled
date. However, users may sometimes need to manually prioritize specific outgoing orders over other
competing demands, especially when existing stock for a product does not adequately fulfill those
demands.

The following documentation covers the process of using an allocation report to manually assign
incoming units of a product from a purchase order (PO) or a manufacturing order (MO) to fulfill an

> |SO| containing the purchased product **OR** an |MO| with the product as a component.

Reviewer 2: I feel like the capitalized, bolded "OR" here kind of gets lost with all the acronyms. Like it becomes just another acronym and not the word "or".

.. _inventory/allocation_report/configuration:

Configuration
=============

By default, allocation reports are automatically enabled for incoming moves (e.g., receipts
generated from |POs|). However, for some operations, users must manually enable the feature.

To do so, navigate to :menuselection:`Inventory app --> Configuration --> Operations Types`. On the
dashboard, click the desired :guilabel:`Operation Type` (e.g., `Receipts` or `Manufacturing`), then
verify that the :guilabel:`Show Allocation` checkbox is enabled. Enabling this option automatically
displays the option to view an :ref:`allocation report
<inventory/allocation_report/workflow/report>` for the specific operation.

.. _inventory/allocation_report/workflow:

Workflow
========

The following outline summarizes all necessary steps for allocating a product using reception
reports:

#. Verify that outgoing demand exists for the product in previously confirmed |SOs| or |MOs|. If

>    not, :ref:`create them <inventory/allocation_report/workflow/confirm_demand>`.

Reviewer 2: I'm not sure if structuring the reference links like this with a fourth part goes against our style guide, but I wanted to call it as something that might get flagged.

   <inventory/allocation_report/workflow/report>` to allocate products.
#. :ref:`Validate the incoming units <inventory/allocation_report/workflow/validate_incoming>` into
   stock.
#. :ref:`Validate the outgoing units <inventory/allocation_report/workflow/validate_outgoing>` to
   fulfill the original demand.

The process begins by ensuring that an outstanding demand for the product already exists. Confirming
a |PO|/|MO| replenishes the stock necessary to meet the demand in the originating |SO|/|MO| either
by purchasing it or by manufacturing it.

On the receipt from the |PO| or on the confirmed |MO|, users can open the allocation report, which
displays the incoming product. Next to the product is displayed a list of all applicable outgoing
transfers (i.e., originating |SOs|/|MOs| containing the product) to which the incoming units can be
assigned. Assigning these units does not move any inventory. Instead, it is a planning method that
allows the user to flexibly choose which demands to prioritize before actually receiving any of the
incoming units. When those units are received, the user can proceed to fulfill the original demands,

> which are correctly reserved.

Reviewer 2: ```suggestion
On the receipt from the |PO| or on the confirmed |MO|, users can open the allocation report, which
displays the incoming product. Next to the product is a list of all applicable outgoing
transfers (i.e., originating |SOs|/|MOs| containing the product) to which incoming units can be
assigned. Assigning these units does not move any inventory. Instead, it
allows the user to flexibly choose which demands to prioritize before actually receiving
incoming units. When those units are received, the user can fulfill the original demands.
```

I thought this sounded a bit more natural overall. Feel free to take or leave as you see fit!

.. note::
   If multiple products are replenished through the |PO|/|MO|, the allocation report lists all
   products with originating |SOs|/|MOs| available for reservation.

.. _inventory/allocation_report/workflow/confirm_demand:

Create and confirm demand
-------------------------

To use allocation reports to assign specific products, a user begins by verifying that previously
confirmed |SOs| exist from customers demanding the product **OR** that confirmed |MOs| exist
requiring the product as a component to be used in production.

> If none exists, create them.

Reviewer 2: I feel like there's something kind of distracting here where the subject is changing between paragraphs. In the first one, we're talking about a generic user, but then we switch to a second-person perspective. I feel like it's sort of obscuring a more direct concept, something like:

Using allocation reports for specific products requires either a previously confirmed SO with the product or an MO where the product is used as a component. If one of two records does not exist, one must be created.

.. important::
   To ensure Odoo can properly create allocations for purchased products, confirm that the selected
   products have the :guilabel:`Purchase` option selected and its :guilabel:`Product Type` set to
   :guilabel:`Goods`.

   For manufactured products, ensure that the product has a :doc:`bill of materials
   <../../../manufacturing/basic_setup/bill_configuration>` configured.

Create |SOs|
~~~~~~~~~~~~

If no confirmed |SO| exists for the product, :doc:`create it
<../../../../sales/sales/sales_quotations/create_quotations>` by opening a new sales quotation form.

> In the order lines, ensure that the requested :guilabel:`Product` and its :guilabel:`Quantity` is

Reviewer 2: ```suggestion
In the order lines, ensure that the requested :guilabel:`Product` and its :guilabel:`Quantity` are
```

specified. Then, confirm the quotation by clicking :guilabel:`Confirm Order`. This turns the
quotation into a confirmed |SO| and generates a delivery order, accessible via the :icon:`fa-truck`
:guilabel:`Delivery` smart button.

Repeat this process if there are multiple customers demanding the product.

Create |MOs|
~~~~~~~~~~~~

If there is no confirmed |MO| with the product as a component, :ref:`create it
<manufacturing/basic_setup/create-mo>` by opening a new |MO| form. In the component lines, add the
requested :guilabel:`Product` and the quantity :guilabel:`To Consume`, then confirm the |MO| by
clicking :guilabel:`Confirm`.

Repeat this process if there are multiple |MOs| requiring the component.

.. _inventory/allocation_report/workflow/confirm_replenish:

Create and confirm |PO| or |MO|
-------------------------------

After confirming existing demand for the product, replenish the necessary stock either by purchasing

> it through |PO| or by manufacturing it through an |MO|.

Reviewer 2: ```suggestion
it through a |PO| or by manufacturing it through an |MO|.
```

Replenish through |PO|
~~~~~~~~~~~~~~~~~~~~~~

> To create a |PO|, navigate to the *Purchase* app and click :guilabel:`New` to create a new Request

Reviewer 2: ```suggestion
To create a |PO|, navigate to the **Purchase** app and click :guilabel:`New` to create a new Request
```

for Quotation (RfQ). On the form, add the products in demand on the customers' |SOs| by clicking
:guilabel:`Add a product` in the *Products* tab and specifying the required :guilabel:`Quantity`.

After adding the products, click :guilabel:`Confirm Order` to turn the |RfQ| into a |PO|.

Replenish through |MO|
~~~~~~~~~~~~~~~~~~~~~~

To create an |MO|, navigate to :menuselection:`Manufacturing app --> Operations --> Manufacturing
Orders`. Then, click :guilabel:`New` to open a new |MO| form. Select the :guilabel:`Product` and
enter the :guilabel:`Quantity` to replenish.

Finally, :guilabel:`Confirm` the |MO|.

.. _inventory/allocation_report/workflow/report:

Allocating using the report
---------------------------

After confirming the replenishment for the product, open the allocation report to assign the pending

> incoming stock to any available originating |SOs|/|MOs| requiring the product.

Reviewer 2: ```suggestion
incoming stock to any available originating |SOs| or |MOs| requiring the product.
```

Feel free to ignore this one. Just a style choice to try and back up the abbreviations. Apply throughout if you dig it.

   |SOs|/|MOs| in the allocation report are listed in chronological order of their associated
   delivery or scheduled date (i.e., the |SO| with the oldest delivery order appears first or the
   |MO| with the oldest scheduled date).

.. image:: allocation_report/allocation-report-page.png
   :alt: Allocation report page in Odoo Inventory.

.. _inventory/allocation_report/workflow/report/allocate:

Allocate units
~~~~~~~~~~~~~~

To allocate units for a product, click the toggle button in the :guilabel:`For` column in the
|SO|/|MO|'s corresponding row. Click the toggle button again to undo the reservation.

Alternatively, to allocate all available units for *all* products across all |SOs|/|MOs|, click
:guilabel:`Assign All` at the top.

The quantity of reservable units for each product is automatically calculated in order, starting
from the oldest |SO|/|MO|. This allows users to allocate as many units as available to fulfill the
quantity specified in the original demand.

.. example::
   For example, |SO| A demands five *Chairs* and |SO| B demands five *Chairs*. A |PO| for eight
   units is created. On the allocation report, |SO| A is listed first with five reservable units.

>    The remaining three available units from this |PO| can be assigned to |SO| B, which will create a

Reviewer 2: ```suggestion
   The remaining three available units from this |PO| can be assigned to |SO| B, which creates a
```

new line showing the allocated :guilabel:`Product` and its allocated :guilabel:`Quantity`.

Finally, click :guilabel:`Validate` to confirm that the products have been delivered to the
customer.

.. image:: allocation_report/delivery-order.png
   :alt: Delivery order from an allocation report.

Fulfill |MO| order
~~~~~~~~~~~~~~~~~~

To fulfill a component demand on an |MO|, navigate to the |MO|. The replenished components in the
previous step are now marked as :guilabel:`Available` for production on this |MO|. Click
:guilabel:`Produce` to manufacture the final product.

.. example::
   The following example demonstrates how an allocation report is used to assign stock for a
   customer |SO| of higher priority.

   Two |SOs| are created for separate customers. Customer A orders 75 units of a *Chair*. Customer B
   also orders 75 units of the same product but needs them as soon as possible.

   No stock exists for the *Chair*, so the user creates and confirms a |PO| to replenish the product
   from Vendor A. This vendor can only provide 100 units of the *Chair* this week. So, the user
   creates and confirms another |PO| to replenish the remaining 50 units from Vendor B, which will

>    be available next week.

Reviewer 2: ```suggestion
   creates and confirms another |PO| to replenish the remaining 50 units from Vendor B next week.
```

Writing around the "will".
